src/validNumberDfa.cpp

```diff
@@ -10,21 +10,29 @@ int isNonZeroDigit(char c) {
         return DS_ACCEPT;
     }
 
+    return DS_ZERO;
+}
+
+int noMoreChars(char c) {
     return DS_ERROR;
 }
 
-int isDigit(char c) {
+int isDigitOrUnderscore(char c) {
```

Comment:
If you go to the Java SE 8 language specification and go to numeric literals you will see some examples that have underscores. 1_2_3 is a valid example and so is 1___2. The language specification outlines more about it